src/libopensc/card-openpgp.c

```diff
@@ -489,23 +489,34 @@ pgp_init(sc_card_t *card)
 
 	/* read information from AID */
 	if (file->namelen == 16) {
+		static char card_name[SC_MAX_APDU_BUFFER_SIZE] = "OpenPGP card";
+
 		/* OpenPGP card spec 1.1, 2.x & 3.x, section 4.2.1 & 4.1.2.1 */
 		priv->bcd_version = bebytes2ushort(file->name + 6);
 		card->version.fw_major = card->version.hw_major = BCD2UCHAR(file->name[6]);
 		card->version.fw_minor = card->version.hw_minor = BCD2UCHAR(file->name[7]);
 
-		/* for "standard" cards, include detailed card version in card name */
+		/* for "standard" cards, include detailed card version & serial no. in card name */
 		if (card->name == default_cardname_v1 ||
 		    card->name == default_cardname_v2 ||
 		    card->name == default_cardname_v3) {
-			static char card_name[SC_MAX_APDU_BUFFER_SIZE] = "OpenPGP card";
-
-			snprintf(card_name, sizeof(card_name), "OpenPGP card v%u.%u",
-				 card->version.hw_major, card->version.hw_minor);
-			card->name = card_name;
+			snprintf(card_name, sizeof(card_name),
+				 "OpenPGP card v%u.%u (%04X %08lX)",
```

Comment:
Why include the serial number in the card name? All other cards keep this seperately; as it is available via `opensc-tool --serial`. Also, I doubt that it will have some meaning for the average user...

Reply:
It _might_ be useful to differentiate between multiple tokens.